py/src/zenbase/predefined/single_class_classifier/classifier.py

```diff
@@ -0,0 +1,134 @@
+__all__ = ["SingleClassClassifier"]
+
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Any, Dict, List, NamedTuple, Optional, Type, TypedDict
+
+from instructor.client import AsyncInstructor, Instructor
+from pydantic import BaseModel
+
+from zenbase.adaptors.json.adaptor import JSONAdaptor
+from zenbase.core.managers import ZenbaseTracer
+from zenbase.optim.metric.labeled_few_shot import LabeledFewShot
+from zenbase.optim.metric.types import CandidateEvalResult
+from zenbase.predefined.base.optimizer import BasePredefinedOptimizer
+from zenbase.predefined.single_class_classifier.function_generator import SingleClassClassifierLMFunctionGenerator
+from zenbase.types import Inputs, LMDemo, LMFunction, Outputs
+
+
+@dataclass(kw_only=True)
+class SingleClassClassifier(BasePredefinedOptimizer):
+    """
+    A single-class classifier that optimizes and evaluates language model functions.
+    """
+
+    class DatasetItem(TypedDict):
+        inputs: Dict[str, Any]
+        outputs: Dict[str, Any]
+
+    class Result(NamedTuple):
+        best_function: LMFunction[Inputs, Outputs]
+        candidate_results: List[CandidateEvalResult]
+        best_candidate_result: Optional[CandidateEvalResult]
+
+    instructor_client: Instructor | AsyncInstructor
+    prompt: str
+    class_dict: Optional[Dict[str, str]] = field(default=None)
+    class_enum: Optional[Enum] = field(default=None)
+    prediction_class: Optional[Type[BaseModel]] = field(default=None)
+    model: str
+    zenbase_tracer: ZenbaseTracer
+    lm_function: Optional[LMFunction] = field(default=None)
+    training_set: List[DatasetItem]
+    test_set: List[DatasetItem]
+    validation_set: List[DatasetItem]
+    shots: int = 5
+    samples: int = 10
+    best_evaluation: Optional[CandidateEvalResult] = field(default=None)
+    base_evaluation: Optional[CandidateEvalResult] = field(default=None)
```

Comment:
Are you able to do the following on Python 3.10?

1. use `| None` instead of `Optional`
2. use `list` instead of `List`